docker/docker_image_dest.go

```diff
@@ -363,6 +365,15 @@ func (d *dockerImageDestination) TryReusingBlobWithOptions(ctx context.Context,
 			remoteName:   reference.Path(candidateRepo),
 			actions:      "pull",
 		}
+		compressionOperation, compressionAlgorithm, err := blobinfocache.OperationAndAlgorithmForCompressor(candidate.CompressorName)
```

Comment:
`extraScope` is only necessary afterwards, so this new check can be performed earlier.

In fact it would be best to add the new check just before the `if candidate.CompressorName != blobinfocache.Uncompressed` line, so that
- all code dealing with `CompressorName` is in a single location
- we don’t log both “trying to reuse” and “ignoring”, only exactly one of the two.